tests/core/structured_vespa_index/test_schemas/healthy_schema_1.sd

```diff
@@ -99,16 +97,16 @@ schema marqo__my_00index {
     }
         rank-profile modifiers inherits default {
             inputs {
-            query(marqo__mult_weights) tensor<float>(p {})
-            query(marqo__add_weights) tensor<float>(p {})
+            query(marqo__mult_weights) tensor<double>(p {})
+            query(marqo__add_weights) tensor<double>(p {})
         }
             function modify(score) {
-            expression: if (count(query(marqo__mult_weights)) == 0, 1, reduce(query(marqo__mult_weights) * attribute(marqo__score_modifiers), prod)) * score + reduce(query(marqo__add_weights) * attribute(marqo__score_modifiers), sum)
+            expression: if (count(query(marqo__mult_weights) * attribute(marqo__score_modifiers_double_long)) == 0,    1, reduce(query(marqo__mult_weights) * attribute(marqo__score_modifiers_double_long), prod)) * if (count(query(marqo__mult_weights) * attribute(marqo__score_modifiers_float)) == 0,    1, reduce(query(marqo__mult_weights) * attribute(marqo__score_modifiers_float), prod)) * score + reduce(query(marqo__add_weights) * attribute(marqo__score_modifiers_double_long), sum) + reduce(query(marqo__add_weights) * attribute(marqo__score_modifiers_float), sum)
```

Comment:
Your rank profile can be simplified to 

if (count(query(marqo__mult_weights)) == 0, 1, reduce(query(marqo__mult_weights) * attribute(marqo__score_modifiers_double_long), prod) * reduce(query(marqo__mult_weights) * attribute(marqo__score_modifiers_float), prod)) * score + reduce(query(marqo__add_weights) * attribute(marqo__score_modifiers_double_long), sum) + reduce(query(marqo__add_weights) * attribute(marqo__score_modifiers_float), sum)

Reply:
Modified to 

```expression = (
                f'(if (count(query({common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS})) == 0, '
                f'    score, '
                f'    if (count(attribute({common.FIELD_SCORE_MODIFIERS_DOUBLE_LONG})) + count(attribute({common.FIELD_SCORE_MODIFIERS_FLOAT})) == 0, '
                f'        0, '
                f'        if (count(query({common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS}) * attribute({common.FIELD_SCORE_MODIFIERS_DOUBLE_LONG})) == 0, '
                f'            1, reduce(query({common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS}) * attribute({common.FIELD_SCORE_MODIFIERS_DOUBLE_LONG}), prod) '
                f'        ) '
                f'        * if (count(query({common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS}) * attribute({common.FIELD_SCORE_MODIFIERS_FLOAT})) == 0, '
                f'            1, reduce(query({common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS}) * attribute({common.FIELD_SCORE_MODIFIERS_FLOAT}), prod) '
                f'        ) '
                f'    ) '
                f')) '
                f'* score '
                f'+ reduce(query({common.QUERY_INPUT_SCORE_MODIFIERS_ADD_WEIGHTS}) * attribute({common.FIELD_SCORE_MODIFIERS_DOUBLE_LONG}), sum) '
                f'+ reduce(query({common.QUERY_INPUT_SCORE_MODIFIERS_ADD_WEIGHTS}) * attribute({common.FIELD_SCORE_MODIFIERS_FLOAT}), sum)'
            )```

Reply:
Reverted to old expression according to team discussion